index.js

```diff
@@ -36,11 +36,154 @@ function hideMobileMenu() {
+      featuredImage: 'https://img.freepik.com/free-photo/selective-focus-african-american-it-employee-with-headset-working-remote-from-home-website-design-using-programming-technologies-computer-programmer-man-coding-digital-business-server_482257-30394.jpg?size=626&ext=jpg&ga=GA1.1.117870262.1687918357&semt=ais',
+      technologies: ['Ruby on Rails', ' CSS', ' JavaScript', ' HTML'],
+      liveVersion: 'https://example.com/project1',
+      sourceLink: 'https://github.com/example/project1',
+    },
+  ];
+
+  function generateProjectDetails(project) {
+    const projectHTML = `
+      <div class="image-container">
+        <img class="featured-image" src="${project.featuredImage}" alt="Project Image">
+      </div>
+      <div class="name-container">
+        <h2 class="project-name">${project.name}</h2>
+        <div class="button-group">
+          <button class="project-button" onclick="window.open('${project.liveVersion}', '_blank')">See live <i class="fas fa-eye"></i></button>
+          <button class="project-button" onclick="window.open('${project.sourceLink}', '_blank')">See source <i class="fab fa-github"></i></button>
+        </div>
+      </div>
+      <ul class="grid-item2">
+        ${project.technologies.map(tech => `<li>${tech}</li>`).join('')}
+      </ul>
+      <div class="project-description">${project.description}</div>
+    `;
+    return projectHTML;
+  }
+
+  function showProjectPopup(projectId) {
+    const project = projects.find(p => p.id === projectId);
+    const projectDetailsContainer = document.getElementById('project-details');
+    projectDetailsContainer.innerHTML = generateProjectDetails(project);
+
+    const popupContainer = document.querySelector('.popup-container');
+    popupContainer.style.display = 'block';
+  }
+
+  function closePopup() {
+    const popupContainer = document.querySelector('.popup-container');
+    popupContainer.style.display = 'none';
+  }
+
+  const seeProjectButtons = document.querySelectorAll('.see-project-button');
+  seeProjectButtons.forEach(button => {
+    button.addEventListener('click', () => {
+      const projectId = parseInt(button.dataset.projectId);
+      showProjectPopup(projectId);
+    });
+  });
+
+  const closeButton = document.querySelector('.close-button');
+  closeButton.addEventListener('click', closePopup);
+});
+
+function getInTouch() {
+  const name = document.getElementById("name").value;
+  const emailInput = document.getElementById("mail");
+  const email = emailInput.value.toLowerCase();
+  const comment = document.getElementById("comment").value;
+
+
+  if (name === "" || email === "" || comment === "") {
+    document.getElementById("errorMessage").innerHTML = "Please fill out all fields.";
+    return;
+  }
+
+  
+  if (email !== emailInput.value.toLowerCase()) {
+    document.getElementById("errorMessage").innerHTML =
+      "Please enter your email address in lowercase letters.";
+    return;
+  }
```

Comment:
- [ ] As mentioned by the previous reviewer please make sure that when a person is submitting an invalid email address with some capital letter in it, the error message is displayed either near the input field or on top of the form fields. You can check if the email is all lowercase by comparing if the email and it's lowercase converted version are the same, ie: `email === email.toLowerCase()`, utilizing `regular expressions`, or looping through each letter of the email string. Any implementation of this check is valid as long as it works properly. 
- [ ] Also please make sure that the form is not submitted if the email is invalid.